cloud-sql/mysql/mysql2/connect-tcp.js

```diff
@@ -0,0 +1,58 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+// [START cloud_sql_mysql_mysql2_connect_tcp]
+// [START cloud_sql_mysql_mysql2_connect_tcp_sslcerts]
+const mysql = require('mysql2/promise');
+const fs = require('fs');
+
+// createTcpSocket initializes a TCP connection pool for a Cloud SQL
+// instance of MySQL.
+const createTcpSocket = async config => {
```

Comment:
oops ... my previous comment said this should be `createTcpSocket` but upon double-checking I think I made a typo and it should be `connectTcpSocket` actually. 😭 My bad!

Ref: https://github.com/GoogleCloudPlatform/golang-samples/blob/main/cloudsql/mysql/database-sql/connect_tcp.go#L35